tests/tools.vitruv.applications.pcmumlclass.tests/src/tools/vitruv/applications/pcmumlclass/tests/SignatureConceptTest.xtend

```diff
@@ -1,200 +1,203 @@
+			val pcmCompositeDataType1 = new FluentPCMCompositeDataTypeBuilder(
+				PcmUmlClassApplicationTestHelper.COMPOSITE_DATATYPE_NAME).build
+			val pcmCompositeDataType2 = new FluentPCMCompositeDataTypeBuilder(
+				PcmUmlClassApplicationTestHelper.COMPOSITE_DATATYPE_NAME_2).build
+			val pcmCollectionDataType = new FluentPCMCollectionDataTypeBuilder(
+				PcmUmlClassApplicationTestHelper.COLLECTION_DATATYPE_NAME, pcmCompositeDataType2).build
+			val expectedPcmRepository = new FluentPCMRepositoryBuilder(PACKAGE_NAME_FIRST_UPPER).addInterface(
+				pcmInterface).addDataType(pcmCompositeDataType1).addDataType(pcmCompositeDataType2).addDataType(
+				pcmCollectionDataType).build
+
+			assertEqualityOfPcmRepository(defaultPcmRepository, expectedPcmRepository)
+		]
 	}
 
-	private def _testReturnTypePropagation_UML(Repository inPcmRepository, Type umlType, int lower, int upper) {
-		var pcmRepository = inPcmRepository
-		var umlInterface = helper.getUmlInterface(pcmRepository)
-		var umlOperation = umlInterface.ownedOperations.head
-		var umlReturnParameter = umlOperation.ownedParameters.findFirst [ param |
-			param.direction === ParameterDirectionKind.RETURN_LITERAL
+	private def _testReturnTypePropagation_UML(Function1<? super Type, Boolean> umlTypeSelector, int lower, int upper,
+		Function1<? super DataType, Boolean> pcmTypeSelector) {
+		changeUmlView[
+			val umlType = it.getRootObjects(Model).flatMap[it.eAllContents.toList].filter[it instanceof Type].map [
+				it as Type
+			].findFirst(umlTypeSelector) as Type
+
+			val umlInterface = UmlQueryUtil.claimInterface(umlContractsPackage,
+				PcmUmlClassApplicationTestHelper.INTERFACE_NAME);
+			var umlOperation = umlInterface.ownedOperations.head
+			var umlReturnParameter = umlOperation.ownedParameters.findFirst [ param |
+				param.direction === ParameterDirectionKind.RETURN_LITERAL
+			]
+
+			umlReturnParameter.type = umlType
+			umlReturnParameter.lower = lower
+			umlReturnParameter.upper = upper
 		]
 
-		umlReturnParameter.type = umlType
-		umlReturnParameter.lower = lower
-		umlReturnParameter.upper = upper
-
-		propagate
-		umlInterface.clearResourcesAndReloadRoot
-		pcmRepository = pcmRepository.clearResourcesAndReloadRoot
-
-		umlInterface = helper.getUmlInterface(pcmRepository)
-		umlOperation = umlInterface.ownedOperations.head
-
-		checkSignatureConcept(umlOperation)
-		var reloadedUmlType = helper.getModifiableInstance(umlType)
-		assertNotNull(reloadedUmlType, "The DataType should not be null after reload")
-		assertTrue(EcoreUtil.equals(umlOperation.type, reloadedUmlType))
-		assertEquals(lower, umlOperation.lower)
-		assertEquals(upper, umlOperation.upper)
-		assertEquals(
-			1,
-			umlOperation.ownedParameters.size,
-			"The Operation should have only a return parameter, since no other parameters were supposed to be added by this test."
-		)
+		validatePcmView[
+			val pcmIntegerDataType = PcmUmlClassApplicationTestHelper.createPrimitiveDataType(PrimitiveTypeEnum.INT)
```

Comment:
Shouldn't this use the pcmTypeSelector? Otherwise, the umlTypeSelector is used but the pcmTypeSelector isn't, which makes it impossible to use another primitive type other than int.